cmd/admin/auth.go

```diff
@@ -102,8 +103,57 @@ func handlerAuthCheck(h http.Handler) http.Handler {
 				samlMiddleware.RequireAccount(h).ServeHTTP(w, r)
 			}
 		case settings.AuthHeaders:
-			// Access always granted
-			h.ServeHTTP(w, r)
+			username := r.Header.Get(headersConfig.TrustedPrefix + headersConfig.UserName)
+			groups := strings.Split(r.Header.Get(headersConfig.TrustedPrefix+headersConfig.Groups), ",")
+		    fullname := r.Header.Get(headersConfig.TrustedPrefix + headersConfig.DisplayName)
+
+			// A username is required to use this system
+			if username == "" {
+				http.Redirect(w, r, forbiddenPath, http.StatusBadRequest)
+				return
+			}
+
+			s := make(contextValue)
+			s["user"] = username
+
+			for _, group := range groups {
+				if group == headersConfig.AdminGroup {
+					s["level"] = adminLevel
+					// We can break because there is no greater permission level
+					break
+				} else if group == headersConfig.UserGroup {
+					s["level"] = userLevel
+					// We can't break because we might still find a higher permission level
+				}
+			}
+
+			// This user didn't present a group that has permission to use the service
+			if _, ok := s["level"]; !ok {
+				http.Redirect(w, r, forbiddenPath, http.StatusForbidden)
+				return
+			}
+
+			if !adminUsers.Exists(username) {
+			  log.Printf("User not found, creating: %s", username)
+			  _, err := adminUsers.New(username, "", fullname, (s["level"] == adminLevel))
```

Comment:
There is one `string` parameter missing, also my bad since I forgot that `.New()` just creates the `AdminUser` structure, and in order to complete the creation of the user, it needs a subsequent call to `.Create()`, for example:
```
email := r.Header.Get(headersConfig.TrustedPrefix + headersConfig.Email)
newUser, err := adminUsers.New(username, "", email, fullname, (s["level"] == adminLevel))
if err != nil {
  log.Printf("Error with new user %s: %v", username, err)
  http.Redirect(w, r, forbiddenPath, http.StatusFound)
  return
}
if err := adminUsers.Create(newUser); err != nil {
  log.Printf("Error creating user %s: %v", username, err)
  http.Redirect(w, r, forbiddenPath, http.StatusFound)
  return
}
```